.github/workflows/ci-testing.yml

```diff
@@ -128,7 +127,7 @@ jobs:
       if: contains( matrix.topic , 'serve' )
       run: |
         sudo apt-get install libsndfile1
-        pip install '.[all,audio]' icevision effdet --upgrade
+        pip install '.[all,audio]' icevision sahi==0.8.19 effdet --upgrade
```

Comment:
why `icevision`, `sahi`, `effdet` are not part of the requirements?

Reply:
They are. We don't need these modules for `serve` though, but for `serve` testing we need them. Hence explicit installation. @Borda

Reply:
```suggestion
        pip install '.[all,audio,serve]' --upgrade
```

Reply:
but still why `all` does not include all?

Reply:
uhmm, because we only need `icevision, effdet` for the serve tests. And these are in `requirements/datatype_image_extras.txt` (which has a lot more than we need for serve tests).

So basically:

- `icevision, effdet` are not needed for serve in Lightning Flash.
- `icevision, effdet` are _only_ needed for "serve tests" in Flash.

Hence the explicit mention of `sahi, icevision, effdet` (sahi because we need to pin it).

Does that help, @Borda ? Please let me know if you have any questions or suggestions.

Reply:
still, `sahi, icevision, effdet` shall be in a requirement file for reproducibility in case you would need to restrict versions

Reply:
They are here: https://github.com/Lightning-AI/lightning-flash/blob/c506d590aebb9e339b159c6aa0d48c97ec1fb733/requirements/datatype_image_extras.txt#L4-L8

And we already use this^ for image.

And this is only temporary, once icevision is fixed for the latest `sahi` release, this will go.

Reply:
let's do these requirements cleaning in another PR, and lets link this discussion  :otter:

Reply:
Yes! ^ From an off-line discussion with @Borda, I fully agree that we should make reproducible environments for users to have the same testing environment as the CI. Nobody will read the CI dependencies to figure out why it fails in the CI.

So, we'll do something similar to what TorchMetrics has: https://github.com/Lightning-AI/metrics/tree/master/requirements

I'll make a PR and link the discussion here. Thank you, @Borda!!